src/main/java/bio/terra/service/job/StairwayLoggingHooks.java

```diff
@@ -15,47 +16,91 @@
         return HookAction.CONTINUE;
     }
 
     @Override
     public HookAction startStep(FlightContext context) {
-        logger.info(StepLogFormat, "startStep", context.getFlightClassName(), context.getFlightId(),
+        MDC.put(FLIGHT_ID_KEY, context.getFlightId());
+        MDC.put(FLIGHT_CLASS_KEY, context.getFlightClassName());
+        MDC.put(FLIGHT_STEP_CLASS_KEY, context.getStepClassName());
+        MDC.put(FLIGHT_STEP_DIRECTION_KEY, context.getDirection().toString());
+        MDC.put(FLIGHT_STEP_NUMBER_KEY, Integer.toString(context.getStepIndex()));
+        MDC.put(FLIGHT_OPERATION_KEY, FLIGHT_STEP_OPERATION_START);
+        logger.info(StepLogFormat, FLIGHT_STEP_OPERATION_START, context.getFlightClassName(), context.getFlightId(),
             context.getStepClassName(), context.getStepIndex(), context.getDirection().name(),
             Instant.now().atZone(ZoneId.of("Z")).format(DateTimeFormatter.ISO_INSTANT));
-        performanceLogger.timerStart("stairwayStep" + context.getFlightId());
+        performanceLogger.timerStart(getStepTimerName(context.getFlightId()));
         return HookAction.CONTINUE;
     }
 
     @Override
     public HookAction endFlight(FlightContext context) {
-        logger.info(FlightLogFormat, "endFlight", context.getFlightClassName(),
+        MDC.put(FLIGHT_OPERATION_KEY, FLIGHT_OPERATION_END);
+        logger.info(FlightLogFormat, FLIGHT_OPERATION_END, context.getFlightClassName(),
             context.getFlightId(), Instant.now().atZone(ZoneId.of("Z")).format(DateTimeFormatter.ISO_INSTANT));
-        performanceLogger.log(context.getFlightId(), context.getFlightClassName(), "endFlight");
+        performanceLogger.log(context.getFlightId(), context.getFlightClassName(), FLIGHT_OPERATION_END);
+        clearMdcKeys();
         return HookAction.CONTINUE;
     }
 
     @Override
     public HookAction endStep(FlightContext context) {
-        logger.info(StepLogFormat, "endStep", context.getFlightClassName(), context.getFlightId(),
+        MDC.put(FLIGHT_OPERATION_KEY, FLIGHT_STEP_OPERATION_END);
+        logger.info(StepLogFormat, FLIGHT_STEP_OPERATION_END, context.getFlightClassName(), context.getFlightId(),
             context.getStepClassName(), context.getStepIndex(), context.getDirection().name(),
             Instant.now().atZone(ZoneId.of("Z")).format(DateTimeFormatter.ISO_INSTANT));
-        performanceLogger.timerEndAndLog("stairwayStep" + context.getFlightId(), context.getFlightId(),
-            context.getFlightClassName(), "endStep", context.getStepIndex());
+        performanceLogger.timerEndAndLog(getStepTimerName(context.getFlightId()), context.getFlightId(),
+            context.getFlightClassName(), FLIGHT_STEP_OPERATION_END, context.getStepIndex());
+        clearMdcKeys();
         return HookAction.CONTINUE;
     }
+
+    private void clearMdcKeys() {
+        MDC.remove(FLIGHT_ID_KEY);
```

Comment:
Are we guaranteed that `endStep()` will be called? If `remove()` is never called, what are the downsides?